mailing_list/listserv_clients/tests.py

```diff
@@ -0,0 +1,282 @@
+import json
+
+from django.test import TestCase
+
+from mock import patch
+
+from icommons_common.utils import Bunch
+
+from .exceptions import ListservApiError
+from .mailgun import MailgunClient as ListservClient
+from mailing_list.models import MailingList
+
+
+class ListservClientTests(TestCase):
+    fixtures = [
+        'mailing_list/fixtures/test_mailing_list.json'
+    ]
+    longMessage = True
+
+    @patch('requests.get')
+    def test_get_list_success(self, mock_requests_get):
+        listserv_client = ListservClient()
+        response = Bunch(
+            status_code=200,
+            json=lambda: {'list': {}}
+        )
+        mock_requests_get.return_value = response
+        mailing_list = MailingList.objects.get(id=1)
+        api_url = "%s/%s" % (listserv_client.api_url, mailing_list.address)
+        auth = (listserv_client.api_user, listserv_client.api_key)
+
+        result = listserv_client.get_list(mailing_list)
+
+        mock_requests_get.assert_called_with(api_url, auth=auth)
+        self.assertEqual(result, {})
+
+    @patch('requests.get')
+    def test_get_list_not_found(self, mock_requests_get):
+        listserv_client = ListservClient()
+        response = Bunch(
+            status_code=404,
+            json=lambda: {}
+        )
+        mock_requests_get.return_value = response
+        mailing_list = MailingList.objects.get(id=1)
+        api_url = "%s/%s" % (listserv_client.api_url, mailing_list.address)
+        auth = (listserv_client.api_user, listserv_client.api_key)
+
+        result = listserv_client.get_list(mailing_list)
+
+        mock_requests_get.assert_called_with(api_url, auth=auth)
+        self.assertIsNone(result)
+
+    @patch('requests.get')
+    def test_get_list_exception(self, mock_requests_get):
+        listserv_client = ListservClient()
+        response = Bunch(
+            status_code=500,
+            text="Call to get_list failed."
+        )
+        mock_requests_get.return_value = response
+        mailing_list = MailingList.objects.get(id=1)
+        api_url = "%s/%s" % (listserv_client.api_url, mailing_list.address)
+        auth = (listserv_client.api_user, listserv_client.api_key)
+
+        with self.assertRaises(ListservApiError):
+            listserv_client.get_list(mailing_list)
+            mock_requests_get.assert_called_with(api_url, auth=auth)
```

Comment:
If the `ListservApiError` is raised by the `listserv_client.get_list()` call, it appears this next line will not be reachable. (Same for other `with self.assertRaises()` blocks in these tests.)